modules/engine/src/paramConverter.ts

```diff
@@ -135,6 +135,9 @@ export async function convertWithdrawParams(
   );
 
   const initiatorSignature = await signer.signMessage(commitment.hashToSign());
+  if (!initiatorSignature) {
+    return Result.fail(new Error("Signer fails to sign message"));
+  }
```

Comment:
We should wrap the whole thing (including the `await` in a try-catch to properly use the `Result` type

Reply:
done.